packages/redux-dynostore-react-redux-subspace/test/mapExtraState.test.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2018, IOOF Holdings Limited.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react'
+import { createStore } from 'redux'
+import { Provider, connect } from 'react-redux'
+import { SubspaceProvider } from 'react-redux-subspace'
+import { mount } from 'enzyme'
+
+import dynostore from '../../redux-dynostore-core/lib/index'
+import dynamic from '@redux-dynostore/react-redux'
+import subspaced from 'src'
+
+describe('mapExtraState tests', () => {
+  let store
+  const initialState = {
+    testId: {
+      testKey: 'root value'
+    },
+    parentId: {
+      testId: {
+        testKey: 'nested value'
+      }
+    },
+    extraState: {
+      extraStateId: 29
+    },
+    invalidExtraState: [ 0, 1, 2 ]
+  }
+
+  class TestComponent extends React.Component {
+    componentDidMount() {
+      this.props.testDispatch()
+    }
+
+    render() {
+      return <p>{this.props.extraStateId}</p>
+    }
+  }
+
+  const mapStateToProps = state => ({
+    testKey: state.testKey,
+    extraStateId: state.extraState.extraStateId
+  })
+
+  const mapDispatchToProps = {
+    testDispatch: () => ({ type: 'TEST_DISPATCH' })
+  }
+
+  const ConnectedTestComponent = connect(mapStateToProps, mapDispatchToProps)(TestComponent)
+
+  beforeEach(() => {
+    store = createStore((state = initialState) => state, dynostore())
+  })
+
+  test('should create subspaced enhanced component with extraState', () => {
+
+    const SubspacedComponent = dynamic('testId', subspaced({
+      mapExtraState: (state, rootState) => ({
+        extraState: rootState.extraState
+      })
+    }))(ConnectedTestComponent)
+
+    const wrapper = mount(
+      <Provider store={store}>
+        <SubspaceProvider mapState={state => state.parentId} namespace="parentId">
+          <SubspacedComponent />
+        </SubspaceProvider>
+      </Provider>
+    )
+
+    expect(wrapper.html()).toBe('<p>29</p>')
+  })
+
+  test('should fail to create subspaced enhanced component with invalid extraState', () => {
+
+    const SubspacedComponent = dynamic('testId', subspaced({
+      mapExtraState: (state, rootState) => (
+        rootState.invalidExtraState
+      )
+    }))(ConnectedTestComponent)
+
+    // Silence error thrown by react
+    console.error = () => {}
```

Comment:
This is a nasty hack but I haven't found a way to silence react error thrown to console.  Even adding and error boundary object did not help.

Reply:
One man's hack is another man's hammer: Exhibit [A](https://github.com/ioof-holdings/redux-subspace/blob/master/packages/redux-subspace-loop/test/namespaced.spec.js#L13), [B](https://github.com/mpeyper/react-purely-stateful/blob/master/test/stateful/stateful.test.js#L6) and [C](https://github.com/mpeyper/react-redux-lifecycle/blob/master/test/withLifecycleActions.test.js#L31)

Reply:
I'm not sure that this will ever reset `console.error` for other tests (possibly hiding useful debugging information)

Reply:
Exhibit B caught my eye